### PR TITLE
feat: move library and enabled.json from CLAUDE_CONFIG_DIR to CLAUDEUP_HOME

### LINKED ISSUE #138: local: move .library and enabled.json from CLAUDE_CONFIG_DIR to CLAUDEUP_HOME
## Problem

claudeup stores its own data (`.library/` directory, `enabled.json`) inside `$CLAUDE_CONFIG_DIR` (`~/.claude/`), which is owned by Claude Code. This creates:

- **Collision risk**: If Claude Code ever introduces a `.library` concept or `enabled.json`, we break
- **Blurred ownership**: Hard to reason about what belongs to claudeup vs Claude Code
- **Backup/cleanup confusion**: Deleting `~/.claude/` would destroy claudeup library data

## Proposal

Move both `.library/` and `enabled.json` from `$CLAUDE_CONFIG_DIR` to `$CLAUDEUP_HOME` (`~/.claudeup/`):

**Before:**
```
~/.claude/.library/agents/foo.md       # claudeup data in Claude's dir
~/.claude/enabled.json                 # claudeup data in Claude's dir
~/.claude/agents/foo.md -> ../.library/agents/foo.md  # relative symlink
```

**After:**
```
~/.claudeup/.library/agents/foo.md     # claudeup data in claudeup's dir
~/.claudeup/enabled.json               # claudeup data in claudeup's dir
~/.claude/agents/foo.md -> /Users/<user>/.claudeup/.library/agents/foo.md  # absolute symlink
```

## Changes Required

- Update `local.Manager` to use `CLAUDEUP_HOME` for `.library/` and `enabled.json`
- Switch symlinks from relative to absolute (necessary since library and active dirs are no longer siblings)
- Update `sync` command to recreate symlinks with absolute paths
- Update all tests
- Update demo script
- Update CLAUDE.md documentation

## Tradeoffs

- Absolute symlinks are slightly less portable than relative ones, but home directories rarely move in practice
- Breaking change for existing setups, but user base is currently just one person
- Cleaner separation of concerns going forward

## Migration

On first run after upgrade, detect old `.library` in `$CLAUDE_CONFIG_DIR` and offer to migrate:
1. Move `.library/` to `$CLAUDEUP_HOME/.library/`
2. Move `enabled.json` to `$CLAUDEUP_HOME/enabled.json`
3. Recreate symlinks with absolute paths via `sync`

### PR DESCRIPTION
**Pull request by malston**
## Summary

Moves claudeup's own data (local items directory and enabled.json) out of `~/.claude/` (Claude Code's config directory) into `~/.claudeup/` (claudeup's own home directory). Switches symlinks from relative to absolute.

Closes #138.

### What changed

- **Two-directory architecture**: `NewManager(claudeDir, claudeupHome)` now takes explicit paths for Claude's active directory and claudeup's storage directory
- **Local items** moved from `~/.claude/.library/` to `~/.claudeup/local/`
- **Config** moved from `~/.claude/enabled.json` to `~/.claudeup/enabled.json`
- **Absolute symlinks** replace relative ones: `~/.claude/agents/foo.md` -> `~/.claudeup/local/agents/foo.md`
- **Doctor command** gains broken symlink detection with recursive scanning via `filepath.WalkDir`
- **Migration script** (`scripts/migrate-library.sh`) handles one-time migration for existing users
- **Terminology**: all code, comments, CLI help text, and docs now use "local storage" instead of "library"

### Why

claudeup was storing its data inside `~/.claude/`, which is Claude Code's directory. This created coupling -- Claude Code updates could interfere with claudeup's data, and vice versa. Moving to `~/.claudeup/` gives claudeup its own namespace while keeping symlinks in `~/.claude/` where Claude Code expects them.

### Files changed

- 34 files changed, 701 insertions, 467 deletions
- Core: `internal/local/` (Manager constructor, config, symlinks, install, list, resolve, view)
- Commands: `internal/commands/` (root, local, doctor, profile, setup, plugin)
- Profile: `internal/profile/` (apply, snapshot -- thread claudeupHome)
- Tests: unit, integration, and acceptance tests updated for two-directory pattern
- Docs: `docs/commands.md`, CLI help text, demo script
- Scripts: `scripts/migrate-library.sh` (new), `scripts/demo-local-management.sh` (updated)

## Test plan

- [x] All unit tests pass (`internal/local/*_test.go`)
- [x] Integration tests pass (`test/integration/local/`, `test/integration/`)
- [x] Acceptance tests pass (`test/acceptance/`)
- [x] Full suite: 18 packages, 0 failures
- [x] Migration script tested end-to-end (moves .library, converts symlinks, handles edge cases)
- [x] `go build ./...` succeeds
- [ ] Manual smoke test: fresh install + enable/disable/import/install cycle
- [ ] Manual smoke test: migration from old .library layout

### CODE REVIEW

File: internal/profile/snapshot_test.go [hunk 1]
```diff
@@ -383,7 +383,7 @@ func TestSnapshotCapturesLocalItems(t *testing.T) {
 	})
 
 	// Create snapshot
-	p, err := Snapshot("test-local-items", claudeDir, claudeJSONPath)
+	p, err := Snapshot("test-local-items", claudeDir, claudeJSONPath, claudeDir)
 	if err != nil {
```
[the remark this answers is not in the document]
Comment (malston) [reply internal/profile/snapshot_test.go]: Good catch. This test is in the profile unit test package -- noted for follow-up. The integration and acceptance level tests now properly separate the directories (2d026ec).
